### PR TITLE
Create CONTRIBUTING.md and UPGRADING.md

### PR DESCRIPTION
**Pull request by NitinAgg**
# Description

# Safety checklist
* [ ] This change is backwards compatible and safe to apply by existing users
* [ ] This change will NOT lead to data loss
* [ ] This change will NOT lead to downtime who already has an env/service setup

## How has this change been tested, beside unit tests?
YOUR_ANSWER

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+**Feature/Large changes:** If you intend to change the public API, or make any non-trivial changes to the implementation, we require you to file a new issue. This lets us reach an agreement on your proposal before you put significant effort into it. You are welcome to submit a PR along with the issue (sometimes necessary for demonstration), but we will not review/merge the code until there is an agreement on the issue.
+
+In general, small PRs are always easier to review than large PRs. The best practice is to break your work into smaller independent PRs and refer to the same issue. This will greatly reduce turnaround time.
+
+If you wish to share your work which is not ready to merge yet, create a [Draft PR](https://github.blog/2019-02-14-introducing-draft-pull-requests/). This will enable maintainers and the CI runner to prioritize mature PR's.
+
+Finally, never submit a PR that will put the main branch in broken state. If the PR is part of multiple PRs to complete a large feature and cannot work on its own, you can create a feature branch and merge all related PRs into the feature branch before creating a PR from feature branch to main.
```
Comment (bigbitbus) [inline CONTRIBUTING.md:67]: Contributors may not know how to test this - whether their PR is breaking the main branch. We should point them to which tests they need to perform before submitting PR (e.g. unit tests, linting with scripts/lint.py etc) , and then which ones will be performed by the core team later (e.g. CircleCI).

Comment (NitinAgg) [reply CONTRIBUTING.md:67]: So we already have a setting that if unit tests etc are failing the PR cannot be merged. I think it's fine to just have this line as a guideline. If someone still sends a PR with broken tests, we can remind them of this. They can't merge it anyway.

**Review by juandiegopalomino — APPROVED** [review]

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+The best way to report a bug is to file an issue on GitHub. Please make sure there is an open issue discussing your contribution. Before opening a new issue, please check for [existing issues](https://github.com/run-x/opta/issues). If you find an existing issue that matches closely with yours, please thumbs-up or comment on it, so we know that the issue is relevant to many people. For any new issue please include:
+
+- Your operating system name and version.
+- Opta version.
+- Detailed steps to reproduce the bug.
+- Any details about your local setup that might be helpful in troubleshooting.
+- When posting Python stack traces, please quote them using
+[Markdown blocks](https://help.github.com/articles/creating-and-highlighting-code-blocks/).
+- Tag the issue with `bug`
+- Remember that this is a volunteer-driven project, and that contributions are welcome :)
```
Comment (RemyDeWolf) [inline CONTRIBUTING.md:27]: all this information and some boiler plate can be in a issue template
ex: https://github.com/DataDog/datadog-agent/blob/main/.github/ISSUE_TEMPLATE.md

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+- For any serious (production breaking) bug we will try to resolve ASAP and do a hotfix release
+- For other bugs we will try to resolve them within the next 2 releases (There is a release every 2 weeks).
+
+### Submit Ideas or Feature Requests
+
+The best way is to file an issue on GitHub:
+
+- Explain in detail how it would work.
+- Keep the scope as narrow as possible, to make it easier to implement.
+- Tag the issue with `feature-request`
```
Comment (RemyDeWolf) [inline CONTRIBUTING.md:41]: Tag -> Label

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+## Orientation
+
+Here's a list of repositories that contain Opta-related code:
+
+- [opta](https://github.com/run-x/opta)
+  is the main repository containing the `Opta` core code and modules.
+- [opta-docs](https://github.com/run-x/opta-docs) contains all the documentation and the Opta installation script.
+
+## Types of Contributions
```
Comment (RemyDeWolf) [inline CONTRIBUTING.md:14]: In the future, we could have a section for custom module here?

File: CONTRIBUTING.md (hunk 2 above)
Comment (NitinAgg) [reply CONTRIBUTING.md:27]: We already have issue templates: https://github.com/run-x/opta/tree/main/.github/ISSUE_TEMPLATE 

I am just repeating things from there. Helps folks who haven't created an issue yet, on what the process would look like when they do.

File: CONTRIBUTING.md (hunk 4 above)
Comment (NitinAgg) [reply CONTRIBUTING.md:14]: 💯 %

File: CONTRIBUTING.md (hunk 2 above)
Comment (RemyDeWolf) [reply CONTRIBUTING.md:27]: cool, sorry I forgot to check

**Review by RemyDeWolf — APPROVED** [review]

**Pull request merged**